### DISCUSSION TITLE
SENEC: State of Health ermitteln

### DISCUSSION THREAD

**Discussion by ledermann** [solectrus/solectrus#3540] [General]
Da @halmich das Thema der Kapazitätsmessung gerade in #3538 aufgeworfen hat, fällt mir dieser kleine Hack dazu ein:

Das BMS eines Akkus ermittelt ja üblicherweise seinen **State of Health (SOH)**. Der SENEC-Stromspeicher macht das auch, die Werte lassen sich über `lala.cgi` auslesen. Die SENEC-App zeigt diese Information aber nicht an, zumindest nicht für meinen V3.

Mit folgendem Statement lässt sich die Akku-Gesundheit eines SENEC-Stromspeichers (Home V3, vermutlich auch V2.1 - aber NICHT Home 4) auslesen:

```bash
curl -s --request POST \
     --url https://192.168.178.29/lala.cgi \
     --data '{ "BMS": { "SOH": "" } }' \
     --insecure \
  | grep -o '"u8_[0-9a-fA-F]*"' \
  | sed 's/"//g' \
  | sed 's/u8_//g' \
  | xargs -I {} printf "State of Health (SOH): %d%%\n" 0x{}
```

(getestet auf Mac, Raspberry Pi und Synology NAS)

Bei meinen V3 mit 10 kWh ergibt das für die vier Module:

```
State of Health (SOH): 99%
State of Health (SOH): 99%
State of Health (SOH): 99%
State of Health (SOH): 99%
```

Allerdings kommen mir diese Werte etwas seltsam vor – nach vier Jahren hätte ich mehr Verschlechterung erwartet. 

Wie ist das bei euch? Wie gesund sind eure Akkus?

**Comment by ledermann** [comment]
Die von @halmich in #3538 beschriebene Kapazitätsanalyse habe ich auch mal gemacht. Ich schreibe das hier, weil es im Issue nur darum gehen sollte, was in SOLECTRUS als Feature ergänzt werden sollte.

Per InfluxDB-Frontend habe ich nach einer Nacht gesucht, in der der Speicher sich maximal (also von 100% auf 0%) entladen hat, ohne zwischendurch zu laden. Eine solche Nacht zu finden, war gar nicht so einfach. Das kommt bei mir (mit den 10 kWh) nämlich recht selten vor - offenbar ein weiterer Hinweis auf einen überdimensionierten Stromspeicher. Im März 2024 bin ich fündig geworden. Das habe ich dann genauer untersucht.

Über eine manuelle InfluxDB-Query ergab sich dann eine entladene Energie von 10,014 kWh. Gegenprobe: Die Beladung im gleichen Zeitraum war etwa 0, also war das tatsächlich nur Entladen. In meinem Fall scheint somit alles ok zu sein, die Nennkapazität lässt sich auch entladen. Passt zum SOH von 99%.

In SOLECTRUS selbst lässt sich das alles leider nicht machen, da man dort nur ganze Tage sehen kann. Hier geht es aber um Nächte. Und mit manuellen InfluxDB-Queries ist das alles recht fummelig.

Zusammengefasst also:

Bei @halmich: Akku mit 5 kWh entlädt 4,75 kWh (Verlust 5 %)
Bei mir: Akku mit 10 kWh entlädt 10,014 kWh (kein Verlust)

Kann man diese Schlussfolgerung auf die aktuelle Kapazität wirklich so machen? Kann ein neuer Akku wirklich 100% der Kapazität am Stück liefern? Könnte man nicht auch argumentieren, dass die beobachtete Differenz von 5% einfach nur Umwandlungsverluste sind? Aber warum gibt es bei mir gar keinen Verlust?

Soweit ich weiß, haben die Akkus intern ja eine größere Kapazität, sonst könnte SENEC nicht 10 Jahre lang die 100% garantieren. Ist also die Reserve bei @halmich bereits aufgebraucht, bei mir aber noch nicht?

**Reply by halmich** [reply]
Ich kann mit InfluxDB leider nicht umgehen, daher kann ich diese Messungen nicht machen.
Allerdings habe ich vorhin mal die CSV Dateien von mein-senec benutzt und beide Rechnungen durchgeführt, hier allerdings mit den 5min Intervall-Werten (tatsächlich ist es bei mir leicht einen Tag zu finden, an dem nur geladen wurde (von 0 auf 100%) und ebenso von 100% auf 0%. Gerade im März und Oktober klappt das gut).

Ergebnis: 
In der Nacht vom 10./11. Oktober wurde die Energie 4,55 kWh abgegeben, es fehlen also schon knapp 9%.
Beladen wurde übrigens 4,8kWh. Das sind 5% Verlust und passt ganz gut zu den Verlust Werte, die sich auch direkt in Solectrus (z.B. in der Jahresübersicht) ausrechnen lassen.

Ich kann mir gut vorstellen, dass mein Speicher die Reserve bereits aufgebraucht hat. Ich habe vermutlich Akkus mit NMC Technologie, du wahrscheinlich NCA (das haben die meisten, ich weiß selbst nicht, wieso ich NMC habe). Vielleicht liegts auch daran.

Im Frühjahr werde ich wohl mal den Installateur um einen offiziellen Kapazitätstest bitten (das lässt sich im Installateurs-Bereich anstoßen) und dann schaue wie Senec reagiert.

**Reply by ledermann** [reply]
Meine Verrenkungen mit InfluxDB entsprechend dem, was du bereits in #3538 gemacht hast. Also Zeitraum auswählen (mit Uhrzeit) und darauf das Integral bilden.

Bei mir ist (laut SOLECTRUS, identisch übrigens mit mein-senec.de) auf die gesamten vier Jahre 1,5% **mehr** entladen worden als beladen. Und ja, bei mir sind NCA-Akkus verbaut.

Einen offiziellen Kapazitätstest machen zu lassen klingt angebracht - hoffentlich wird dabei mehr gemacht als mein obiges Auslesen der SOH-Werte ;-)

**Reply by halmich** [reply]
Evtl. sind es bei dir nur 1,5%, da du den SenecV3 hast. Kann es sein, dass hier die Batterie direkt per DC vom Dach geladen wird?
Ich hab den v2.1, der einfach per AC angeschlossen ist, und daher die Energie erst noch in DC umwandeln muss. Das geht nicht verlustfrei.

**Reply by ledermann** [reply]
Bei meinem Speicher ist es genau so, wie du sagst:

> Der SENEC.Home V3 hybrid verfügt über einen integrierten trafolosen Wechselrichter, durch den Ihre Anlage nur einen statt zwei Wechselrichter benötigt. Der „Hybridspeicher“ wandelt den Gleichstrom aus der PV-Anlage für die unmittelbare Nutzung oder Einspeisung in Wechselstrom um, oder er speichert den Gleichstrom und wandelt ihn erst bei der späteren Abgabe an den Haushalt in Wechselstrom um. Das reduziert nur Umwandlungsverluste und spart Ihnen die Anschaffungs- und Installationskosten für einen separaten Wechselrichter.
Quelle: https://senec.com/de/produkte/senec-home/v3

Übrigens sind es bei mir nicht 1,5% Verlust, sondern Gewinn ;-) Auf die bislang vier Jahre gesehen wurden 1,5% **mehr** entnommen, als hineingesteckt wurden. Naja, vermutlich Messungenauigkeiten.

**Reply by halmich** [reply]
Ja verrückt! wusste noch, dass ich kurz verwirrt war bei den Zahlen, aber mir war nicht klar was die Verwirrung ausgelöst hat.
das wars 😀
du hast Energie aus dem Nichts erzeugt 😀

**Reply by ledermann** [reply]
... wenn mein alter Physik-LK-Lehrer das noch hätte miterleben dürfen! ;-)

**Comment by halmich** [comment]
> Wie ist das bei euch? Wie gesund sind eure Akkus?

Bei mir ebenfalls 99% auf beiden Modulen.

**Comment by gereons** [comment]
Die neuen Ampace-Akkus schreiben ihren SoH übrigens direkt ins Logfile: 

> 2024-10-13 16:07:40 [I|Ampace] Module #B [Balancing][Cellvoltage]: [ ...] SoC: 50% **SOH: 100%** Current: -25.6A Module-Limits: 60.0A/-60.0A Temps: 21, 22, 24C Voltage: 52.0V System-Limits: 0.0A/-29.5A LV: -52.0A

**Comment by bennysweb** [comment]
> Wie ist das bei euch? Wie gesund sind eure Akkus?

Bei mir ebenfalls alle 4 Module 99% SOH nach 2,5 Jahren Betrieb. Ist also scheinbar nicht wirklich aussagekräftig ;-)
